sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/parsers/LineCountParser.java

```diff
@@ -38,46 +39,75 @@
  */
 public final class LineCountParser extends AbstractParser {
 
+  private final CommentHandler commentHandler;
+
   private static class CommentHandler extends DefaultHandler implements LexicalHandler {
 
-    private int lastLineNumberCode = 0;
-    private int lastLineNumberComment = 0;
+    private Deque<Integer> commentLines = new ArrayDeque<>();
+    private Deque<Integer> effectiveCommentLines = new ArrayDeque<>();
+
+    private int lastCodeLine = 0;
 
     private Locator locator;
-    private int numCommentLines;
 
     private void registerLineOfCode() {
-      lastLineNumberCode = locator.getLineNumber();
-      if (lastLineNumberComment >= lastLineNumberCode) {
-        numCommentLines--;
+      lastCodeLine = locator.getLineNumber();
+      if (lastEffectiveCommentLine() == lastCodeLine) {
+        effectiveCommentLines.pop();
       }
+      if (lastCommentLine() == lastCodeLine) {
+        commentLines.pop();
+      }
+    }
+
+    private int lastEffectiveCommentLine() {
+      return effectiveCommentLines.isEmpty() ? 0 : effectiveCommentLines.peek();
+    }
+
+    private int lastCommentLine() {
+      return commentLines.isEmpty() ? 0 : commentLines.peek();
     }
 
     public void comment(char[] ch, int start, int length) throws SAXException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1161)